src/main/java/org/ow2/proactive/workflow_catalog/rest/util/ArchiveManagerHelper.java

```diff
@@ -0,0 +1,137 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.workflow_catalog.rest.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.ow2.proactive.workflow_catalog.rest.entity.WorkflowRevision;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class ArchiveManagerHelper {
+
+    /**
+     * Get the name of the Workflow used into the archive (in order not to have twice the same
+     * name). If the name of the Workflow already exists, an index is append to the name.
+     * @param existingNames the set of names already used
+     * @param currentName the Workflow current name
+     * @return the name of the Workflow different from all existing names
+     */
+    public String getName(Set<String> existingNames, String currentName) {
+
+        if (!existingNames.contains(currentName))
+            return currentName;
+
+        for (int i = 1; i <= existingNames.size(); i++) {
+            String newName = currentName + "_" + i;
+            if (!existingNames.contains(newName)) {
+                return newName;
+            }
+        }
+
+        return currentName + "_0";
+
+    }
+
+    /**
+     * Compress a list of WorkflowRevision into a ZIP archive
+     * @param workflowsList the list of workflows to compress
+     * @return a byte array corresponding to the archive containing the workflows
+     */
+    public byte[] compressZIP(List<WorkflowRevision> workflowsList) {
+
+        if (workflowsList == null || workflowsList.size() == 0) {
+            return null;
+        }
+
+        try {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream);
+
+            Set<String> existingNames = new HashSet<>();
+            for (WorkflowRevision file : workflowsList) {
+                String fileName = getName(existingNames, file.getName());
+                existingNames.add(fileName);
+
+                byte[] xmlByteArray = file.getXmlPayload();
+
+                ZipEntry entry = new ZipEntry(fileName);
+                entry.setSize(xmlByteArray.length);
+
+                zipOutputStream.putNextEntry(entry);
+                zipOutputStream.write(xmlByteArray);
+                zipOutputStream.closeEntry();
+            }
+
+            zipOutputStream.close();
```

Comment:
Should close in finally block or use [Try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).